Mathlib/CategoryTheory/Sites/Closed.lean

```diff
@@ -0,0 +1,332 @@
+
+/-- The closure of a sieve is closed. -/
+theorem close_isClosed {X : C} (S : Sieve X) : J₁.IsClosed (J₁.close S) :=
+  fun _ g hg => J₁.arrow_trans g _ S hg fun _ hS => hS
+#align category_theory.grothendieck_topology.close_is_closed CategoryTheory.GrothendieckTopology.close_isClosed
+
+/-- The sieve `S` is closed iff its closure is equal to itself. -/
+theorem isClosed_iff_close_eq_self {X : C} (S : Sieve X) : J₁.IsClosed S ↔ J₁.close S = S := by
+  constructor
+  · intro h
+    apply le_antisymm
+    · intro Y f hf
+      rw [← J₁.covers_iff_mem_of_closed h]
+      apply hf
+    · apply J₁.le_close
+  · intro e
+    rw [← e]
+    apply J₁.close_isClosed
+#align category_theory.grothendieck_topology.is_closed_iff_close_eq_self CategoryTheory.GrothendieckTopology.isClosed_iff_close_eq_self
+
+theorem close_eq_self_of_isClosed {X : C} {S : Sieve X} (hS : J₁.IsClosed S) : J₁.close S = S :=
+  (J₁.isClosed_iff_close_eq_self S).1 hS
+#align category_theory.grothendieck_topology.close_eq_self_of_is_closed CategoryTheory.GrothendieckTopology.close_eq_self_of_isClosed
+
+/-- Closing under `J` is stable under pullback. -/
+theorem pullback_close {X Y : C} (f : Y ⟶ X) (S : Sieve X) :
+    J₁.close (S.pullback f) = (J₁.close S).pullback f := by
+  apply le_antisymm
+  · refine' J₁.le_close_of_isClosed (Sieve.pullback_monotone _ (J₁.le_close S)) _
+    apply J₁.isClosed_pullback _ _ (J₁.close_isClosed _)
+  · intro Z g hg
+    change _ ∈ J₁ _
+    rw [← Sieve.pullback_comp]
+    apply hg
+#align category_theory.grothendieck_topology.pullback_close CategoryTheory.GrothendieckTopology.pullback_close
+
+@[mono]
+theorem monotone_close {X : C} : Monotone (J₁.close : Sieve X → Sieve X) :=
+  fun _ S₂ h => J₁.le_close_of_isClosed (h.trans (J₁.le_close _)) (J₁.close_isClosed S₂)
+#align category_theory.grothendieck_topology.monotone_close CategoryTheory.GrothendieckTopology.monotone_close
+
+@[simp]
+theorem close_close {X : C} (S : Sieve X) : J₁.close (J₁.close S) = J₁.close S :=
+  le_antisymm (J₁.le_close_of_isClosed le_rfl (J₁.close_isClosed S))
+    (J₁.monotone_close (J₁.le_close _))
+#align category_theory.grothendieck_topology.close_close CategoryTheory.GrothendieckTopology.close_close
+
+/--
+The sieve `S` is in the topology iff its closure is the maximal sieve. This shows that the closure
+operator determines the topology.
+-/
+theorem close_eq_top_iff_mem {X : C} (S : Sieve X) : J₁.close S = ⊤ ↔ S ∈ J₁ X := by
+  constructor
+  · intro h
+    apply J₁.transitive (J₁.top_mem X)
+    intro Y f hf
+    change J₁.close S f
+    rwa [h]
+  · intro hS
+    rw [eq_top_iff]
+    intro Y f _
+    apply J₁.pullback_stable _ hS
+#align category_theory.grothendieck_topology.close_eq_top_iff_mem CategoryTheory.GrothendieckTopology.close_eq_top_iff_mem
+
+/-- A Grothendieck topology induces a natural family of closure operators on sieves. -/
+-- @[simps (config := { rhsMd := semireducible })] Porting note: no semireducible
```

Comment:
```suggestion
```
We're making this replacement everywhere, and not leaving a porting note. I think it's okay to do so here, too.